### ISSUE TITLE
There's a error related with nodeproxy

### ISSUE THREAD

**Issue by judoaseeta** [neo4j-graphql/neo4j-graphql-js#37]
I installed apoc3.3.0.2 on my local neo4j 3.3.4 Enterprise database.

then i add @cypher directive on my graphql schema  in order to make a subquery.

and i got this error message.
"Failed to invoke function `apoc.cypher.runFirstColumn`: Caused by: java.lang.IllegalArgumentException: Can not be converted to long: org.neo4j.kernel.impl.core.NodeProxy

here is my schema definition.

```
type ART {
        title: String!
        artId: Int!
        similarArts(first: Int = 3, offset: Int = 0):[ART] @cypher(statement: "MATCH (this)-[:TYPE_OF]->(:TYPE)<-[:TYPE_OF]-(o:ART) RETURN o")
    }
    type Query {
        getArtsByTitle(title: String): [ART]
        getArtsById(artId: Int): [ART]
    }
```

is this error related with graphql or some jvm related issues?(I'm not Java guy.)

**Comment by judoaseeta** [comment]
It is solved. since i replace relations '[:TYPE_OF]' as plain '--'. it works.

**Comment by frankwanicka** [comment]
I ran into the same thing, and it occurs running it directly in Neo4j browser as well, so it's not related to this library. I saw some code samples for runFirstColumn that showed the directional arrow, so apparently it should be possible, but currently it bombs for me.

**Comment by johnymontana** [comment]
Thanks for reporting @judoaseeta and for confirming @frankwanicka. I'm able to duplicate this as well and it only seems to occur when using a Cypher query with a directed relationship. Seems to be a Neo4j procedure specific issue that needs to resolved in the APOC library:

https://github.com/neo4j-contrib/neo4j-apoc-procedures/issues/776

A similar issue exists in the neo4j-graphql plugin: https://github.com/neo4j-graphql/neo4j-graphql/issues/98

**Comment by johnymontana** [comment]
We've determined that this is an error related to the compiled runtime in Neo4j and how these subqueries are executed using the `runFirstColumn` procedure. Specifically, it has to do with passing node parameters when the compiled runtime is enabled.

The reason some queries work and some result in this error is that the compiled runtime is not supported for all queries. As a temporary workaround until this is fixed in Neo4j you can disable compiled runtime by setting

```
unsupported.cypher.runtime=SLOTTED
```

in neo4j.conf

**Comment by johnymontana** [comment]
This should now be resolved in recent versions of Neo4j (3.4.1+) and APOC. 
If you run into this, please upgrade. Closing the issue, but if you experience again please reopen.
